codegen/smithy-kotlin-codegen/src/main/kotlin/software/amazon/smithy/kotlin/codegen/model/RulesEngineExt.kt

```diff
@@ -42,5 +43,8 @@ fun Value.toLiteral(): String =
     when (this) {
         is StringValue -> value.doubleQuote()
         is BooleanValue -> value.toString()
+        is ArrayValue -> values.joinToString(",", "mutableListOf(", ")") { value ->
+            value.expectStringValue().value.doubleQuote()
+        }
```

Comment:
**Nit**: This general pattern of serializing node values to a string of double-quoted values is repeated multiple times. We should commonize it into a single function (perhaps an extension method on `List<Value>`).